src/UpdatedPackagesCollector.js

```diff
@@ -157,7 +157,7 @@ export default class UpdatedPackagesCollector {
       return (
         this.updatedPackages[pkg.name] ||
         (this.options[SECRET_FLAG] ? false : this.dependents[pkg.name]) ||
-        this.options.canary
+        (this.options.canary && typeof this.options.since !== "string")
```

Comment:
(**Just my opinion**): This condition does not depend on the `pkg` argument (the subject of the filter) so it would be much clearer to understand how this condition affects `collectUpdates` if it was taken out of the `filter` callback. If I'm understanding this correctly, when `canary` is specified and `since` isn't, `collectUpdates` will collect **all** packages. This would be much clearer if written something like:

```js
collectUpdates() {
  const shouldCollectAllPackages = this.options.canary && typeof this.options.since !== 'string';
  let collectedPackages = this.packages;
  
  if (!shouldCollectAllPackages) {
    collectedPackages = this.packages.filter((pkg) => {
      return this.updatedPackages[pkg.name] || this.options[SECRET_FLAG] ? false : this.dependents[pkg.name];
    });
  }

  /* Omitted */
}
```